src/behaviours/seekSafestFood.ts

```diff
@@ -0,0 +1,77 @@
+import { ISnake, IBoard, ICoordinate, Directions, Matrix } from '../Types';
+import Pathfinder from '../Pathfinder';
+import { firstToFood } from '../helpers';
+import { chaseEnemyTail } from './chaseEnemyTail';
+
+/**
+ * @param {Pathfinder} PF - Pathfinder class initialized with game state
+ * @param {IBoard} board - the board state
+ * @param {ISnake} us - our snake
+ * @returns {Directions} returns the next direction
+ */
+export const seekSafestFood = (
+  PF: Pathfinder,
+  board: IBoard,
+  us: ISnake
+): Directions => {
+  try {
+    const snakes: ISnake[] = board.snakes;
+    const head: ICoordinate = us.body[0];
+    const PF = new Pathfinder(board, snakes);
+    const { food: foodArray } = board;
+
+    let pathToSafestFood: Matrix;
+    let safestFood: ICoordinate;
+    // For each food item we check:
+    // 1. Are we closer than any other snake.
+    // 2. After we eat the food do we have a path to another snakes tail (escape route)
+    // If so, we chonk.
+    foodArray.forEach(snakeSnack => {
+      const winnerWinnerChickenDinner = firstToFood(us, snakes, snakeSnack, PF);
+      const pathToSnack = PF.getFullPath(head, snakeSnack);
+      const futureBody = [];
+      // map where our body will be when we eat food.
+      for (const coordinate of pathToSnack) {
+        let futureBodyPoint = {};
+        futureBodyPoint['x'] = coordinate[0];
+        futureBodyPoint['y'] = coordinate[0];
+        futureBody.push(futureBodyPoint);
```

Comment:
Rather than use these declarations, I think we could just do
```
futureBody.push({ x: coordinate[0], y: coordinate[1] });
```

Reply:
Much better 👍